doc/DOCGUIDE.md

```diff
@@ -0,0 +1,88 @@
+
+### Documentation Structure
+
+KallistiOS uses Doxygen to generate documentation from the project's header 
+files and the following .dox files:
+
+- attribution.dox 
+- audio.dox
+- debugging.dox
+- filesystem.dox
+- math.dox
+- networking.dox
+- peripherals.dox
+- system.dox
+- threading.dox
+- timing.dox
+- video.dox
+
+All of them can be found in the doc folder.
+
+#### Topics
+
+Doxygen's documentation is organized in a tree-like structure constructed around
+Topics and Files. In KallistiOS, the previously mentioned list of files describes
+the top level Topics used to group the corresponding subtopics. As an example,
+the audio Topic hosts the Driver, Sound Effects and Streaming subtopics.
+Topics can link to Files and other topics as well. 
+
+#### Files
+
+Files represent (optional) units of documentation that exist under Topics. Doxygen
+typically uses them to document specific source or header files. In KallistiOS,
+Files are used to list macros, function signatures, function lists and the
+headers used in the file.
+
+---
+
+### Documentation Level of Detail
+
+Different levels of detail are provided in KallistiOS' documentation as a way to
+help the reader quickly find their way through the documentation and the OS.
+To achieve these two goals, documentation should be written with more
+detail as it gets closer to the File level.
+
+Topic level documentation should aim to inform the reader of what they will find
+inside it. A top level Topic description must state the API's purpose and
+hardware it might relate to.
+
+Subtopics are the place to start providing more detailed insights into the API
+by describing its capabilities and limitations, providing function descriptions 
+and alerting the reader to common pitfalls and known issues. Subtopics that
+aren't parents to other subtopics should also provide a basic guide into using 
+the API, with a focus on any setup required to use it and its key
+functions.
+
+At the File level, the documentation should focus on describing the inner
+workings of a module as well as more advanced API usecases and data structures.
+
+#### Linking to Other Pages
+
+When expanding the documentation, it is good practice to provide links to
+important functions or macros mentioned. If a Topic has any type of relatioship
+with resources that belong to another Topic, Doxygen functionalities should be
+used to ensure that the page points to the mentioned resources. 
+This will help readers reach important pages that might be of their interest.
+
+---
+
+### Where to Write What
+
+Documentation should be kept in the header files of KallistiOS, while the .dox
+files should be used to definegroups. As modern IDEs are capable of parsing
+Doxygen style documentation, keeping the majority of the documentation inside
+header files can prove useful for users as they use KallistiOS' APIs.
+
+In cases where a module has a top level header and then headers specific to
```

Comment:
So everything here looks great, and I'm glad you've taken the time to do this, but this last part right here I'm having a little bit of trouble conceptualizing. Can you give an example?

Like in the new PVR API, we have `pvr.h` which includes sub-API headers for `pvr_prim.h`, `pvr_scene.h`, etc. Are you saying to somehow add the shit in these sub-headers to get indexed under the main `pvr_prim.h`? I'm not actually positive what the command/mechanism to do that in Doxygen even is, tbh. lol.

Reply:
You got me second guessing myself now. I think I managed to do it with a Doxygen command... but I can't find the place I did it. Might be on my desktop, will have to check later. 

But yes, `pvr_prim.h` should have, in my mind, the minimum information required for the reader to understand how the most important functions 'sub-headers' are used and how different mechanisms relate with one another. Or at least notes to point the reader to the sub-headers telling them what part of the module they will find in each file, and doing it through Doxygen command would save a huge amount of copy/pasting and header reorganization.

The romfs API was my point of reference when writing this part, as the actual description is currently in 'fs_rodisk.h' and not in the topic. But other pages, like the audio driver page probably should get a similar treatment - general description on the top level header, details in the sub-headers.

I didn't include the Doxygen commands yet because I think it's something that needs to be tested and then discussed with other contributors.